Remove server styles host

alan-agius4 opened the pull request:
**refactor(platform-browser): combine `DomSharedStylesHost` and `SharedStylesHost`**
    
The mentioned 2 classes have been combined since it is no longer required to have a separate `SharedStylesHost` for SSR. This changes also reduces the memory usage footprint as remove 1 Map that stores the CSS strings.

---

**refactor(platform-server): remove `ServerStylesHost`**
    
With the recent changes and refactors of `dom_renderer` having a separate `ServerStylesHost` is redundant.

---

**refactor(platform-browser): rename `ng-app` style attribute to `ng-app-id`**

`ng-app` is an AngularJS attribute, see https://docs.angularjs.org/api/ng/directive/ngApp. Using this attribute on a non AngularJS element can cause DI issues in AngularJS when running an AngularJS and Angular application on the same page.

As such, we avoid such problems the Angular `ng-app` attribute is renamed to `ng-app-id`.

alan-agius4 committed:
refactor(platform-server): remove `ServerStylesHost`

With the recent changes and refactors of `dom_renderer` having a separate `ServerStylesHost` is redundant.

alan-agius4 committed:
refactor(platform-browser): combine `DomSharedStylesHost` and `SharedStylesHost`

The mentioned 2 classes have been combined since it is no longer required to have a separate `SharedStylesHost` for SSR. This changes also reduces the memory usage footprint as remove 1 Map that stores the CSS strings.

alan-agius4 committed:
fixup! refactor(platform-browser): combine `DomSharedStylesHost` and `SharedStylesHost`

AndrewKushnir commented on packages/platform-browser/src/dom/shared_styles_host.ts:
```diff
@@ -6,12 +6,28 @@
+  private hostNodes = new Set<Node>();
+  private styleNodesInDOM: Map<string, HTMLStyleElement>|null;
+  private platformIsServer: boolean;
+
+  constructor(
+      @Inject(DOCUMENT) private readonly doc: Document, @Inject(APP_ID) private appId: string,
+      @Inject(PLATFORM_ID) platformId: object = {}) {
```
We can use this change as an opportunity to switch to the `inject` function, WDYT?
```suggestion
  private readonly doc = inject(DOCUMENT);
  private readonly appId = inject(APP_ID);
  platformId = inject(PLATFORM_ID); // should it also be private and readonly?
  private readonly isPlatformServer = isPlatformServer(this.platformId);

  constructor() {
```

alan-agius4 replied:
I was going to do that, but in some cases, this classes is initialised using the ctor ex: https://github.com/angular/angular/pull/49424/files#diff-bd84358b34fb4defc3222724ec52e393a302cb7d3a3df8282415ae8f53a3c7d6R43, so it would to be changed to 

```ts
  doc: Document;

  constructor(
      doc?: Document,
      appId?: string,
      platformId?: object
    ) {

    this.doc = doc ?? inject(DOCUMENT);
    ...
  }
```

Do you still prefer `inject`?

AndrewKushnir replied:
Thanks for the comment. No worries, let's keep things as is then. The mentioned usage comes from the tests and would require some additional refactoring (create an injector and run a function within its context to get a hold of the `SharedStylesHost` token), which we can do afterwards (anytime, after v16).

AndrewKushnir reviewed (approved):
@alan-agius4 thanks for the refactoring! 👍 The changes look great!

alan-agius4 committed:
fixup! refactor(platform-browser): combine `DomSharedStylesHost` and `SharedStylesHost`

alan-agius4 commented:
CL for failing target in G3: http://cl/517104783

alan-agius4 committed:
refactor(platform-browser): rename `ng-app` style attribute to `ng-app-id`

`ng-app` is an AngularJS attribute, see https://docs.angularjs.org/api/ng/directive/ngApp. Using this attribute on a non AngularJS element can cause DI issues in AngularJS when running an AngularJS and Angular application on the same page.

As such, we avoid such problems the Angular `ng-app` attribute is renamed to `ng-app-id`.

AndrewKushnir reviewed (approved):
👍

pkozlowski-opensource commented:
This PR was merged into the repository by commit 9636910c0158f4e99fb0d684d79791159d2d7940.
